internal/gtfs/direction_calculator.go

```diff
@@ -0,0 +1,131 @@
+package gtfs
+
+import (
+	"context"
+	"maglev.onebusaway.org/gtfsdb"
+	"maglev.onebusaway.org/internal/utils"
+)
+
+type DirectionCalculator struct {
+	queries *gtfsdb.Queries
+}
+
+func NewDirectionCalculator(queries *gtfsdb.Queries) *DirectionCalculator {
+	return &DirectionCalculator{
+		queries: queries,
+	}
+}
+
+// CalculateStopDirection determines the compass direction for a stop
+func (dc *DirectionCalculator) CalculateStopDirection(ctx context.Context, stopID string) string {
```

Comment:
not right now, but down the road we should add support for caching these stop direction calculations.